File: FlexServ Backend/SpringBoot/AuthService/src/main/java/com/flexserv/entity/ServiceProviderCompany.java
package com.flexserv.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "service_provider_company")
public class ServiceProviderCompany {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "company_name", nullable = false, length = 150)
    private String companyName;

    @Column(name = "experience_years")
    private Integer experienceYears;

    @Column(columnDefinition = "TEXT")
    private String bio;

    @Column(name = "is_verified")
    private Boolean isVerified = true;

    @Column(name = "rating")
    private Double rating = 5.0;

    @Column(name = "company_available")
    private Boolean companyAvailable = true;
}

File: FlexServ Backend/DotNet/AuthService/Models/ServiceProviderCompany.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AuthService.Models
{
    [Table("service_provider_company")]
    public class ServiceProviderCompany
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [Column("user_id")]
        public long UserId { get; set; }

        [ForeignKey("UserId")]
        public User? User { get; set; }

        [Required]
        [MaxLength(150)]
        [Column("company_name")]
        public string CompanyName { get; set; } = string.Empty;

        [Column("experience_years")]
        public int? ExperienceYears { get; set; }

        [Column("bio")]
        public string? Bio { get; set; }

        [Column("is_verified")]
        public bool? IsVerified { get; set; } = false;

        [Column("rating")]
        public double? Rating { get; set; } = 0.0;

        [Column("company_available")]
        public bool? CompanyAvailable { get; set; } = true;

        public ICollection<ServiceEntity> Services { get; set; } = new List<ServiceEntity>();
    }
}
